lib/ui/screens/event_screen.dart

```diff
@@ -552,41 +552,40 @@ class _EventScreenState extends State<EventScreen> {
         final messenger = ScaffoldMessenger.of(context);
         final calendarCubit = BlocProvider.of<CalendarCubit>(context);
         final router = GoRouter.of(context);
-        var confirmed = !event.cancelDeadlinePassed();
-        if (!confirmed) {
-          confirmed = await showDialog<bool>(
-                context: context,
-                builder: (context) {
-                  return AlertDialog(
-                    title: const Text('Register'),
-                    content: Text(
-                      'Are you sure you want to register? The '
-                      'cancellation deadline has already passed.',
-                      style: Theme.of(context).textTheme.bodyMedium,
-                    ),
-                    actions: [
-                      TextButton.icon(
-                        onPressed: () => Navigator.of(
-                          context,
-                          rootNavigator: true,
-                        ).pop(false),
-                        icon: const Icon(Icons.clear),
-                        label: const Text('NO'),
-                      ),
-                      ElevatedButton.icon(
-                        onPressed: () => Navigator.of(
-                          context,
-                          rootNavigator: true,
-                        ).pop(true),
-                        icon: const Icon(Icons.check),
-                        label: const Text('YES'),
-                      ),
-                    ],
-                  );
-                },
-              ) ??
-              false;
-        }
+        bool confirmed =
+            (event.cancelDeadlinePassed() && event.registration!.isInvited)
```

Comment:
This doesn't make sense to me: "do you want to register" if already registered and invited?

Reply:
thats a very good point, this whole section is kinda a mess so I misinterpreted the code while looking at it at a glance